Dockerfile

```diff
@@ -0,0 +1,24 @@
+﻿FROM mcr.microsoft.com/dotnet/sdk:3.1-cbl-mariner1.0 AS build
+WORKDIR /app
+COPY . .
+RUN dotnet publish -c Release -o out \
+    -r linux-x64 \
+    -p:MinVerSkip=true \
+    --self-contained true \
+    -p:PublishReadyToRun=false \
+    -p:IncludeNativeLibrariesForSelfExtract=true \
+    -p:PublishSingleFile=true \
+    -p:PublishTrimmed=true \
+    -p:TrimUnusedDependencies=true \
+    ./src/Microsoft.ComponentDetection
+
+FROM mcr.microsoft.com/dotnet/runtime-deps:3.1-cbl-mariner1.0 AS runtime
+WORKDIR /app
+COPY --from=build /app/out ./
+
+RUN tdnf install -y \
+    golang \
+    moby-engine \
+    python
```

Comment:
In addition to these, for all detectors to work we'll need to have `gradle`, `maven`, `pnpm` and `poetry` to be installed.